src/custom-hooks/check-visibility.hook.tsx

```diff
@@ -0,0 +1,30 @@
+import { createRef, useEffect, useState } from "react";
```

Comment:
Benefit of naming it "custom-hooks/<name>.hook.tsx"?

Can it just be "custom-hooks/<name>.tsx"? (since it's already in custom-hooks folder)